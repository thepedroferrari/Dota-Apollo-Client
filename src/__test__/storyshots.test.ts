import initStoryshots, {
  Stories2SnapsConverter,
} from '@storybook/addon-storyshots';
import { act, create } from 'react-test-renderer';

const wait = () =>
  act(
    () =>
      new Promise(resolve => {
        setTimeout(resolve, 1000);
      }),
  );

const converter = new Stories2SnapsConverter();

const runTest = async (story: any, context: any) => {
  const filename = converter.getSnapshotFileName(context);

  if (!filename) {
    return;
  }

  const storyElement = story.render();
  let tree;
  act(() => {
    tree = create(storyElement);
  });
  if (!tree) return;

  await wait();
  expect(tree.toJSON()).toMatchSpecificSnapshot(filename);

  tree.unmount();
};

initStoryshots({
  asyncJest: true,
  test: ({ story, context, done }) => {
    runTest(story, context).then(done);
  },
});
