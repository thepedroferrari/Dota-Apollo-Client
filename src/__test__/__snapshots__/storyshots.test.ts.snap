// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Countdown In 1 Week 1`] = `
<time>
  7
  Days, 
  0
  h 
  0
  m 
  0
  s
</time>
`;

exports[`Storyshots Countdown In 5 Minutes 1`] = `
<time>
  0
  Days, 
  0
  h 
  5
  m 
  0
  s
</time>
`;

exports[`Storyshots Countdown My Birthday in 2020 1`] = `
<time>
  229
  Days, 
  7
  h 
  48
  m 
  39
  s
</time>
`;

exports[`Storyshots Countdown My Birthday in 2021 1`] = `
<time>
  594
  Days, 
  7
  h 
  48
  m 
  39
  s
</time>
`;

exports[`Storyshots Countdown Tomorrow 1`] = `
<time>
  1
  Days, 
  0
  h 
  0
  m 
  0
  s
</time>
`;
